### PR TITLE
Update H36M 3d keypoint evaluation metrics

### PR DESCRIPTION
**Pull request by ly015**
* Add n-mpjpe metric
* Remove align_root option. Always relativize the prediction and the groundtruth instead.
* Refactor keypoint_mpjpe function

### CODE REVIEW

**Review by cherryjm — APPROVED** [review]

File: mmpose/core/evaluation/pose3d_eval.py [hunk 1]
```diff
@@ -17,6 +17,12 @@ def keypoint_mpjpe(pred, gt, mask):
         mask (np.ndarray[N, K]): Visibility of the target. False for invisible
             joints, and True for visible. Invisible joints will be ignored for
             accuracy calculation.
+        alignment (str, optional): method to align the prediction with the
+            groundtruth. Supported options are:
+            - ``'none'``: no alignment will be applied
+            - ``'scale'``: align in the least-square scene in scale
```
Comment (innerlee) [inline mmpose/core/evaluation/pose3d_eval.py:23]: What is a "in the least-square scene"

**Review by innerlee — APPROVED** [review]

**Comment by innerlee** [comment]
This is better

File: mmpose/core/evaluation/pose3d_eval.py (hunk 1 above)
Comment (innerlee) [reply mmpose/core/evaluation/pose3d_eval.py:23]: typo of "sense", i guess

File: mmpose/datasets/datasets/body3d/body3d_h36m_dataset.py [hunk 2]
```diff
@@ -230,23 +236,29 @@ def _report_mpjpe(self, keypoint_results, align_root=True):
...
 
-        if align_root:
-            # The prediction and groundtruth are aligned by subtracting root
-            # position respectively (equivalent to move the prediction to the
-            # groundtruth root position by translation). Then theroot joint
-            # will be removed from the pose.
-            preds = preds[:, 1:, :] - preds[:, :1, :]
-            gts = gts[:, 1:, :] - gts[:, :1, :]
-            masks = masks[:, 1:]
+        # Zero-center the pose around its root joint and the remove the root
```
Comment (jin-s13) [inline mmpose/datasets/datasets/body3d/body3d_h36m_dataset.py:239]: ~the~ remove

**Pull request merged**